src/main/java/org/apache/commons/lang3/reflect/TypeUtils.java

```diff
@@ -1811,8 +1828,9 @@ private static String parameterizedTypeToString(final ParameterizedType p) {
         return buf.toString();
     }
 
-    private static void appendRecursiveTypes(final StringBuilder buf, final int[] recursiveTypeIndexes, final Type[] argumentTypes) {
-        for (int i = 0; i < recursiveTypeIndexes.length; i++) {
+    @SuppressWarnings({"compound.assignment.type.incompatible","array.access.unsafe.high"}) // #1 recursiveTypeIndexes.length <= argumentTypes.indexes => i++ from 0 to recursiveTypeIndexes.length has i @IndexOrHigh("argumentTypes")
+    private static void appendRecursiveTypes(final StringBuilder buf, final int @LTEqLengthOf("#3") [] recursiveTypeIndexes, final Type[] argumentTypes) {
```

Comment:
`@LTEqLengthOf` applies to indices, not arrays. See the comment in `findRecursiveTypes`.